### PR TITLE
Makes the donut box craftable and foldable

### PR DESCRIPTION
**Pull request by dovydas12345**
**What does this PR do:**
Allows the player to create donut boxes using cardboard. It also allows to fold the donut box back to cardboard.

**Changelog:**
:cl: Dovydas12345
add: Added the ability to fold and create donut boxes using cardboard.
/:cl:

### CODE REVIEW

**Comment by Citinited** [comment]
Sweet

**Comment by Citinited** [comment]
The only thing I would recommend is to make a new storage-level proc e.g. `obj/item/storage/proc/fold_flat()`, move all the vars to storage level, then put this code in it, then call it when you `attack_self`. Then change the code elsewhere (e.g. on boxes) to call that proc. This will save a couple lines and it's generally good practice to have a proc instead of duplicated code

**Comment by Citinited** [comment]
(Leading on from above, sorry can't edit) Or if you're feeling up to it, add a component that gives this behaviour; or repath donut boxes (and maybe other "fancy" storage things too) to be subtypes of actual boxes / a "foldable" storage path. Just some ideas :)

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 1]
```diff
@@ -48,13 +48,39 @@
...
+	if(!foldable)
+		return
+	if(contents.len)
+		to_chat(user, "<span class='warning'>You can't fold this box with items still inside!</span>")
+		return
+	if(!ispath(foldable))
+		return
+
+	// Close any open UI windows first
+	var/found = 0
```
Comment (Ty-Omaha) [inline code/game/objects/items/weapons/storage/fancy.dm]: booleans should be `TRUE `or `FALSE`

**Comment by dovydas12345** [comment]
I might do something like that later, but i cannot do it this weekend, since i am not home. When i get back, I might be able to do something.

**Comment by variableundefined** [comment]
Given that all box subtype are foldable, I'd suggest changing the subtype of the fancy boxes to be under box and make the necessary accomodations too.

For some reason aquatic storage kit is in this file. It does not belong.

(Not your fault, you are not required to do anything about the latter)

**Comment by dovydas12345** [comment]
Will make a brand new PR for it.

**Pull request closed without merging**